plugins/out_s3/s3.c

```diff
@@ -83,6 +83,18 @@ int s3_plugin_under_test()
     return FLB_FALSE;
 }
 
+struct flb_aws_header create_canned_acl_header(char *canned_acl)
+{
+    struct flb_aws_header h = {
+        .key = "x-amz-acl",
+        .key_len = 9,
+        .val = canned_acl,
+        .val_len = flb_sds_len(canned_acl),
+    };
+
+    return h;
+};
```

Comment:
I'm surprised this code works. I believe you need to `flb_malloc` the memory for the header so its created on the heap, otherwise it's created on the stack, and may be overwritten after this function call.

Reply:
Thanks.. I will check it and make a code change.